forge/ee/lib/sso/index.js

```diff
@@ -323,8 +324,27 @@ module.exports.init = async function (app) {
                         }
                     }
                 }
+                if (providerOpts.groupAdmin && providerOpts.groupAdminName === ga) {
+                    adminGroup = true
+                }
             })
 
+            if (user.admin && !adminGroup) {
```

Comment:
We should only be touching admin if `providerOpts.groupAdmin` is set - otherwise we have to leave the flag alone.

As far as I can tell, this logic, if `groupAdmin` is false, will have left `adminGroup` set to `false` and the next piece will then strip them of the admin flag... unless that is what the desiredTeamMemberships check is handling? Would be good to make it a bit clearer (via more explicit logic, or comments).